core/src/main/java/edu/ucr/cs/riple/core/explorers/DownStreamDependencyExplorer.java

```diff
@@ -162,6 +164,26 @@ public int effectOnDownstreamDependencies(Fix fix) {
     return optional.map(methodStatus -> methodStatus.effect).orElse(0);
   }
 
+  /**
+   * Returns the lower bound of number of errors of applying a fix and its associated chain of fixes
+   * on the target on downstream dependencies.
+   *
+   * @param root Root of the fix tree.
+   * @param chain Chain of the fix tree associated to root.
+   * @return Lower bound of number of errors on downstream dependencies.
+   */
+  public int computeLowerBoundOfNumberOfErrors(Fix root, Set<Fix> chain) {
```

Comment:
Can we also compute the upper bound (sum) and record both values somewhere, even if we don't use the upper bound to compute the full cost right now? I kinda want to know how close to real effect on downstream deps this is, and the easiest way to gain some certainty that is close would be to see that the lower and upper bound are usually close.

Reply:
da4be8dd8867bad27678eb9b090135370502cca0

Sounds great :), implemented in this PR.

Reply:
Took a quick look at the additional commit, LGTM :)